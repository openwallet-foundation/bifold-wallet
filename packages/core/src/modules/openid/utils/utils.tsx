import { getDomainFromUrl } from '@credo-ts/core'
import { Attribute, Field } from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../types'

/**
 * Converts a camelCase string to a sentence format (first letter capitalized, rest in lower case).
 * i.e. sanitizeString("helloWorld")  // returns: 'Hello world'
 */
export function sanitizeString(str: string) {
  const result = str.replace(/([a-z0-9])([A-Z])/g, '$1 $2').replaceAll('_', ' ')
  let words = result.split(' ')
  words = words.map((word, index) => {
    if (index === 0 || word.toUpperCase() === word) {
      return word.charAt(0).toUpperCase() + word.slice(1)
    }
    return word.charAt(0).toLowerCase() + word.slice(1)
  })
  return words.join(' ')
}

export function getHostNameFromUrl(url: string) {
  try {
    return getDomainFromUrl(url)
  } catch (error) {
    throw new Error(`Error getting hostname from url: ${error}`)
  }
}

export const buildFieldsFromOpenIDTemplate = (data: { [key: string]: unknown }): Array<Field> => {
  const fields = []
  for (const key of Object.keys(data)) {
    // omit id and type
    if (key === 'id' || key === 'type') continue

    let pushedVal: string | number | null = null
    if (typeof data[key] === 'string' || typeof data[key] === 'number') {
      pushedVal = data[key] as string | number | null
    }
    fields.push(new Attribute({ name: key, value: pushedVal }))
  }
  return fields
}

export function formatDate(input: string | Date): string {
  const date = input instanceof Date ? input : new Date(input)
  return date.toLocaleString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    hour12: true,
  })
}

export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): Attribute | undefined => {
  for (const [key, value] of Object.entries(display.attributes)) {
    if (searchKey === key) {
      return new Attribute({
        name: key,
        value: value as string | number | null,
        mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
      })
    }
  }
  return undefined
}

/**
 * Mostly used for mdoc crednetials
 */
export function detectImageMimeType(data: Uint8Array): 'image/jpeg' | 'image/jp2' | null {
  // Check if array is too short to contain magic numbers
  if (data.length < 12) {
    return null
  }

  // JPEG starts with FF D8 FF
  if (data[0] === 0xff && data[1] === 0xd8 && data[2] === 0xff) {
    return 'image/jpeg'
  }

  // JPEG2000 has two possible signatures:
  // 1) 00 00 00 0C 6A 50 20 20 0D 0A 87 0A
  // 2) FF 4F FF 51

  // Check first signature
  if (
    data[0] === 0x00 &&
    data[1] === 0x00 &&
    data[2] === 0x00 &&
    data[3] === 0x0c &&
    data[4] === 0x6a && // 'j'
    data[5] === 0x50 && // 'P'
    data[6] === 0x20 &&
    data[7] === 0x20 &&
    data[8] === 0x0d &&
    data[9] === 0x0a &&
    data[10] === 0x87 &&
    data[11] === 0x0a
  ) {
    return 'image/jp2'
  }

  // Check second signature
  if (data[0] === 0xff && data[1] === 0x4f && data[2] === 0xff && data[3] === 0x51) {
    return 'image/jp2'
  }

  return null
}

/**
 * very simple matcher for `yyyy-mm-dd`
 */
export function isDateString(value: string) {
  // We do the length check first to avoid unnecesary regex
  return value.length === 'yyyy-mm-dd'.length && value.match(/^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])$/)
}

export function isW3CProofRequest(type: string): boolean {
  if (type === 'jwt_vc') {
    return true
  }
  return false
}

export function isSdJwtProofRequest(type: string): boolean {
  if (type === 'vc+sd-jwt') {
    return true
  }
  return false
}
