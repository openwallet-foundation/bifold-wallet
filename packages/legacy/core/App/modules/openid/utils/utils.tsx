import { getDomainFromUrl } from '@credo-ts/core'
import { Attribute, Field } from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../types'

/**
 * Converts a camelCase string to a sentence format (first letter capitalized, rest in lower case).
 * i.e. sanitizeString("helloWorld")  // returns: 'Hello world'
 */
export function sanitizeString(str: string) {
  const result = str.replace(/([a-z0-9])([A-Z])/g, '$1 $2').replaceAll('_', ' ')
  let words = result.split(' ')
  words = words.map((word, index) => {
    if (index === 0 || word.toUpperCase() === word) {
      return word.charAt(0).toUpperCase() + word.slice(1)
    }
    return word.charAt(0).toLowerCase() + word.slice(1)
  })
  return words.join(' ')
}

export function getHostNameFromUrl(url: string) {
  try {
    return getDomainFromUrl(url)
  } catch (error) {
    throw new Error(`Error getting hostname from url: ${error}`)
  }
}

export const buildFieldsFromOpenIDTemplate = (data: { [key: string]: unknown }): Array<Field> => {
  const fields = []
  for (const key of Object.keys(data)) {
    // omit id and type
    if (key === 'id' || key === 'type') continue

    let pushedVal: string | number | null = null
    if (typeof data[key] === 'string' || typeof data[key] === 'number') {
      pushedVal = data[key] as string | number | null
    }
    fields.push(new Attribute({ name: key, value: pushedVal }))
  }
  return fields
}

export function formatDate(input: string | Date): string {
  const date = input instanceof Date ? input : new Date(input)
  return date.toLocaleString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    hour12: true,
  })
}

export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): Attribute | undefined => {
  for (const [key, value] of Object.entries(display.attributes)) {
    if (searchKey === key) {
      return new Attribute({
        name: key,
        value: value as string | number | null,
        mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
      })
    }
  }
  return undefined
}
