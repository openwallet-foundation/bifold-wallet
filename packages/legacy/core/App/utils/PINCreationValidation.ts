import { PINValidationRules } from '../types/security'

const consecutiveSeriesOfThree = new RegExp(/012|123|234|345|456|567|678|789|987|876|765|654|543|432|321|210/)
const evenNumberSeries = new RegExp('(13579)')
const oddNumberSeries = new RegExp('(02468)')
const isNumber = new RegExp('^[0-9]+$')
const crossNumberPattern = ['159753', '159357', '951357', '951753', '357159', '357951', '753159', '753951']

export enum PINError {
  CrossPatternValidation = 'CrossPatternValidation',
  OddOrEvenSequenceValidation = 'OddOrEvenSequenceValidation',
  NoRepetitionOfTheSameNumbersValidation = 'NoRepetitionOfTheSameNumbersValidation',
  MaxAdjacentNumberRepetitionValidation = 'MaxAdjacentNumberRepetitionValidation',
  NoRepetitionOfTheTwoSameNumbersValidation = 'NoRepetitionOfTheTwoSameNumbersValidation',
  NoSeriesOfNumbersValidation = 'NoSeriesOfNumbersValidation',
  PINOnlyContainDigitsValidation = 'PINOnlyContainDigitsValidation',
  PINTooShortValidation = 'PINTooShortValidation',
  PINTooLongValidation = 'PINTooLongValidation',
}

export interface PINValidationsType {
  isInvalid: boolean
  errorName: PINError
}

export const PINCreationValidations = (PIN: string, PINRules: PINValidationRules) => {
  const PINValidations: PINValidationsType[] = []
  
  if (PINRules.no_cross_pattern) {
    PINValidations.push({
      isInvalid: crossNumberPattern.includes(PIN),
      errorName: PINError.CrossPatternValidation,
    } as PINValidationsType)
  }
  if (PINRules.no_even_or_odd_series_of_numbers) {
    PINValidations.push({
      isInvalid: evenNumberSeries.test(PIN) || oddNumberSeries.test(PIN),
      errorName: PINError.OddOrEvenSequenceValidation,
    } as PINValidationsType)
  }

  if (1 == PINRules.no_repeated_numbers) {
    const repetitionPattern = new RegExp(/(\d)\1{1,}/)
    PINValidations.push({
      isInvalid: repetitionPattern.test(PIN),
      errorName: PINError.NoRepetitionOfTheSameNumbersValidation
    } as PINValidationsType)
  } else if (1 < PINRules.no_repeated_numbers){
    const repetitionPattern = new RegExp(String.raw`(\d)\1{${PINRules.no_repeated_numbers},}`, "g")
    PINValidations.push({
      isInvalid: repetitionPattern.test(PIN),
      errorName: PINError.MaxAdjacentNumberRepetitionValidation
    } as PINValidationsType)
  }

  if (PINRules.no_repetition_of_the_two_same_numbers) {
    let noRepetitionOfTheTwoSameNumbers = new RegExp(/([0-9][0-9])\1{1,}/)
    if (typeof PINRules.no_repetition_of_the_two_same_numbers === 'number') {
      noRepetitionOfTheTwoSameNumbers = new RegExp(
        `([0-9][0-9])\\1{${PINRules.no_repetition_of_the_two_same_numbers - 1},}`
      )
    }
    PINValidations.push({
      isInvalid: noRepetitionOfTheTwoSameNumbers.test(PIN),
      errorName: PINError.NoRepetitionOfTheTwoSameNumbersValidation,
    } as PINValidationsType)
  }
 
  if (PINRules.no_series_of_numbers) {
    PINValidations.push({
      isInvalid: consecutiveSeriesOfThree.test(PIN),
      errorName: PINError.NoSeriesOfNumbersValidation,
    } as PINValidationsType)
  }
  if (PINRules.only_numbers) {
    PINValidations.push({
      isInvalid: !isNumber.test(PIN),
      errorName: PINError.PINOnlyContainDigitsValidation,
    } as PINValidationsType)
  }

  PINValidations.push({
    isInvalid: PIN.length < PINRules.min_length || PIN.length > PINRules.max_length,
    errorName: PIN.length <= PINRules.max_length ? PINError.PINTooShortValidation : PINError.PINTooLongValidation,
  } as PINValidationsType)

  return PINValidations
}
