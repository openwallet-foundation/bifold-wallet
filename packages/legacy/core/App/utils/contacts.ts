import { BasicMessageRecord, ConnectionRecord, CredentialExchangeRecord, ProofExchangeRecord } from '@credo-ts/core'

import { BifoldAgent } from './agent'

interface ConnectionWithMessages {
  conn: ConnectionRecord
  msgs: (BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord)[]
}

interface ConnectionWithLatestMessage {
  conn: ConnectionRecord
  latestMsg: BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord
}

/**
 * Function to fetch contacts (connections) in order of latest chat message without using hooks
 * @param agent - Credo agent
 * @returns ConnectionRecord[] sorted by most recent message
 */
export const fetchContactsByLatestMessage = async (agent: BifoldAgent, connections: ConnectionRecord[]): Promise<ConnectionRecord[]> => {
  const connectionsWithMessages = await Promise.all<ConnectionWithMessages>(
    connections.map(
      async (conn: ConnectionRecord): Promise<ConnectionWithMessages> => ({
        conn,
        msgs: [
          ...(await agent.basicMessages.findAllByQuery({ connectionId: conn.id })),
          ...(await agent.proofs.findAllByQuery({ connectionId: conn.id })),
          ...(await agent.credentials.findAllByQuery({ connectionId: conn.id })),
        ],
      })
    )
  )

  const connectionsWithLatestMessage: ConnectionWithLatestMessage[] = connectionsWithMessages.map((pair) => {
    return {
      conn: pair.conn,
      latestMsg: pair.msgs.reduce(
        (acc, cur) => {
          const accDate = acc.updatedAt || acc.createdAt
          const curDate = cur.updatedAt || cur.createdAt
          return accDate > curDate ? acc : cur
        },
        // Initial value if no messages exist for this connection is a placeholder with the date the connection was created
        { createdAt: pair.conn.createdAt } as BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord
      ),
    }
  })

  return connectionsWithLatestMessage
    .sort(
      (a, b) =>
        new Date(b.latestMsg.updatedAt || b.latestMsg.createdAt).valueOf() -
        new Date(a.latestMsg.updatedAt || a.latestMsg.createdAt).valueOf()
    )
    .map((pair) => pair.conn)
}
