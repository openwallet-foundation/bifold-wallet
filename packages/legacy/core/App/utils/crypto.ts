import argon2 from 'react-native-argon2'

export const hashPIN = async (PIN: string, salt: string): Promise<string> => {
  try {
    const result = await argon2(PIN, salt, {})
    const { rawHash } = result

    return rawHash
  } catch (error) {
    throw new Error(`Error generating hash for PIN ${String((error as Error)?.message ?? error)}`)
  }
}
