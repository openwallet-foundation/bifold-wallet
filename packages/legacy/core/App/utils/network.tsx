import TcpSocket from 'react-native-tcp-socket'

import pools from '../configs/ledgers/indy'
import { GenesisTransaction } from '../types/genesis'

export const canConnectToHost = async (host: { host: string; port: number }): Promise<boolean> =>
  new Promise((resolve) => {
    const socketTimeoutInMs = 3000
    const client = TcpSocket.createConnection(host, () => {
      resolve(true)
      client.destroy()
    })

    // Other events that can be safely be ignored. See the
    // library for more details:
    // https://www.npmjs.com/package/react-native-tcp-socket

    client.on('error', () => {
      client.destroy()
      resolve(false)
    })

    client.on('timeout', () => {
      client.destroy()
      client.removeAllListeners()

      resolve(false)
    })

    client.setTimeout(socketTimeoutInMs)
  })

export const fetchLedgerNodes = (indyNamespace = 'sovrin'): Array<{ host: string; port: number }> => {
  const [pool] = pools.filter((p) => p.indyNamespace === indyNamespace)
  if (!pool) {
    return []
  }

  const genesisTransactionsAsString = pool.genesisTransactions
  let genesisTransactions: Array<GenesisTransaction> = []

  if (genesisTransactionsAsString) {
    try {
      genesisTransactions = genesisTransactionsAsString.split('\n').map((g) => JSON.parse(g))
    } catch (error: unknown) {
      return []
    }
  }

  const nodes = genesisTransactions.map((g) => {
    return { host: g.txn.data.data.client_ip, port: parseInt(g.txn.data.data.client_port) }
  })

  return nodes
}
