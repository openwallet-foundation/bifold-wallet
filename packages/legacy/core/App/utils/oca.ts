import { CredentialExchangeRecord } from '@credo-ts/core'
import { AnonCredsProofRequestTemplatePayloadData, CredentialSharedProofData } from '@hyperledger/aries-bifold-verifier'
import {
  Attribute,
  CredentialOverlay,
  Field,
  OCABundleResolveAllParams,
  OCABundleResolverType,
  Predicate,
} from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../modules/openid/types'
import { BrandingOverlay } from '@hyperledger/aries-oca'

export const buildFieldsFromAnonCredsCredential = (credential: CredentialExchangeRecord): Array<Field> => {
  return credential?.credentialAttributes?.map((attr) => new Attribute(attr)) || []
}

export const buildFieldsFromW3cCredsCredential = (value: W3cCredentialDisplay): Array<Field> => {
  return (
    Object.entries(value.attributes).map(
      ([key, value]) =>
        new Attribute({
          name: key,
          value: value as string | number | null,
          mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
        })
    ) || []
  )
}

export const buildFieldsFromAnonCredsProofRequestTemplate = (
  data: AnonCredsProofRequestTemplatePayloadData
): Array<Field> => {
  const fields = []
  if (data.requestedAttributes) {
    for (const item of data.requestedAttributes) {
      if (item.name) {
        fields.push(new Attribute({ name: item.name, value: null, ...item }))
      }
      if (item.names) {
        for (const name of item.names) {
          fields.push(new Attribute({ name, value: null, ...item }))
        }
      }
    }
  }
  if (data.requestedPredicates) {
    for (const item of data.requestedPredicates) {
      fields.push(
        new Predicate({
          pType: item.predicateType,
          pValue: item.predicateValue,
          ...item,
        })
      )
    }
  }
  return fields
}

export const buildFieldsFromSharedAnonCredsProof = (data: CredentialSharedProofData): Array<Field> => {
  const fields = []
  for (const attribute of data.sharedAttributes) {
    fields.push(
      new Attribute({
        name: attribute.name,
        value: attribute.value,
      })
    )
  }
  for (const attributesGroup of data.sharedAttributeGroups) {
    for (const attribute of attributesGroup.attributes) {
      fields.push(
        new Attribute({
          name: attribute.name,
          value: attribute.value,
        })
      )
    }
  }
  for (const predicate of data.resolvedPredicates) {
    fields.push(
      new Predicate({
        name: predicate.name,
        pType: predicate.predicateType,
        pValue: predicate.predicateValue,
        satisfied: true,
      })
    )
  }
  return fields
}

export const buildOverlayFromW3cCredential = async ({
  credentialDisplay,
  language,
  resolver,
}: {
  credentialDisplay: W3cCredentialDisplay
  language: string
  resolver: OCABundleResolverType
}): Promise<CredentialOverlay<BrandingOverlay>> => {
  const params: OCABundleResolveAllParams = {
    identifiers: {
      schemaId: '',
      credentialDefinitionId: credentialDisplay.id,
    },
    meta: {
      alias: credentialDisplay.display.issuer.name,
      credConnectionId: undefined,
      credName: credentialDisplay.display.name,
    },
    attributes: buildFieldsFromW3cCredsCredential(credentialDisplay),
    language,
  }

  const bundle = await resolver.resolveAllBundles(params)

  return {
    ...(bundle as CredentialOverlay<BrandingOverlay>),
    presentationFields: buildFieldsFromW3cCredsCredential(credentialDisplay),
  }
}
