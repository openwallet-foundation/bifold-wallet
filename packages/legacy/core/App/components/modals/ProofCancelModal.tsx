import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { ScrollView, StyleSheet, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import { useTheme } from '../../contexts/theme'
import { GenericFn } from '../../types/fn'
import { testIdWithKey } from '../../utils/testable'
import Button, { ButtonType } from '../buttons/Button'
import { ThemedText } from '../texts/ThemedText'
import SafeAreaModal from './SafeAreaModal'

interface ProofCancelModalProps {
  visible?: boolean
  onDone?: GenericFn
}

const ProofCancelModal: React.FC<ProofCancelModalProps> = ({ visible, onDone }) => {
  const { t } = useTranslation()
  const { ColorPallet, Assets } = useTheme()

  const styles = StyleSheet.create({
    safeAreaView: {
      backgroundColor: ColorPallet.brand.modalPrimaryBackground,
      flex: 1,
    },
    container: {
      flexGrow: 1,
      padding: 20,
    },
    controlsContainer: {
      marginHorizontal: 20,
      marginBottom: 10,
    },
    content: {
      justifyContent: 'space-around',
    },
    heading: {
      marginTop: 60,
      marginBottom: 30,
      textAlign: 'center',
    },
    image: {
      alignSelf: 'center',
      marginBottom: 50,
    },
  })

  return (
    <SafeAreaModal visible={visible} animationType="slide">
      <SafeAreaView style={styles.safeAreaView}>
        <ScrollView style={styles.container}>
          <View style={styles.content}>
            <ThemedText variant="modalTitle" style={styles.heading}>
              {t('ProofRequest.NoInfoShared')}
            </ThemedText>
            <Assets.svg.noInfoShared style={styles.image} height={200} />
            <ThemedText variant="modalNormal" style={{ textAlign: 'center' }}>
              {t('ProofRequest.YourInfo')}
            </ThemedText>
          </View>
        </ScrollView>
        <View style={styles.controlsContainer}>
          <View style={{ paddingTop: 10 }}>
            <Button
              title={t('Global.Done')}
              accessibilityLabel={t('Global.Done')}
              testID={testIdWithKey('Done')}
              onPress={onDone}
              buttonType={ButtonType.ModalPrimary}
            />
          </View>
        </View>
      </SafeAreaView>
    </SafeAreaModal>
  )
}

export default ProofCancelModal
