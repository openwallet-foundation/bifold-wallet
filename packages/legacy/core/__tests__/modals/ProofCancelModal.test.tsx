import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import ProofCancelModal from '../../App/components/modals/ProofCancelModal'
import { testIdWithKey } from '../../App/utils/testable'

describe('ProofCancelModal Component', () => {
  test('Rerenders correctly when not visible', async () => {
    const tree = render(<ProofCancelModal visible={true} onDone={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })

  test('Done button triggers callback', async () => {
    const onDone = jest.fn()
    const tree = render(<ProofCancelModal onDone={onDone} visible={true} />)

    const doneButton = tree.getByTestId(testIdWithKey('Done'))

    fireEvent(doneButton, 'press')

    expect(onDone).toBeCalledTimes(1)
  })
})
