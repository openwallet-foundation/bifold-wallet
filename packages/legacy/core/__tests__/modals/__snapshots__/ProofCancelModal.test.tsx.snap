// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProofCancelModal Component Rerenders correctly when not visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  visible={true}
>
  <RNCSafeAreaView
    edges={
      Object {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      Object {
        "backgroundColor": "#000000",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "flexGrow": 1,
          "padding": 20,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "justifyContent": "space-around",
            }
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "marginBottom": 30,
                  "marginTop": 60,
                  "textAlign": "center",
                },
              ]
            }
          >
            ProofRequest.NoInfoShared
          </Text>
          <
            height={200}
            style={
              Object {
                "alignSelf": "center",
                "marginBottom": 50,
              }
            }
          />
          <Text
            maxFontSizeMultiplier={2}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            ProofRequest.YourInfo
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginHorizontal": 20,
        }
      }
    >
      <View
        style={
          Object {
            "paddingTop": 10,
          }
        }
      >
        <View
          accessibilityLabel="Global.Done"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/Done"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    false,
                    false,
                    false,
                  ],
                ]
              }
            >
              Global.Done
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
